src/Devices/EthernetServer_TC.cpp

```diff
@@ -178,56 +199,59 @@ void EthernetServer_TC::post() {
   } else if (!file()) {
     // TODO: send an error response
     serial(F("put \"%s\" not recognized!"), buffer + 5);
-    client.stop();
-    state = NOT_CONNECTED;
+    state = FINISHED;
   }
 }
 
 void EthernetServer_TC::loop() {
+  if (state == FINISHED) {  // Tear down
+    state = NOT_CONNECTED;
+    memset(buffer, 0, sizeof(buffer));
+    bufferContentsSize = 0;
+    connectedAt = 0;
+    client.stop();
+    return;
```

Comment:
It’s not at the end of the function, since there’s possibly code that can be executed after that block. I could make it an else-if statement afterward though.

Reply:
No, leave as is. I wasn't reading carefully.